File: WorkQueues/python/sender/src/new_task.py
import pika
import sys

QUEUE_NAME = 'task_queue'

connection = pika.BlockingConnection(
    pika.ConnectionParameters(host='rabbitmq')
)

channel = connection.channel()

channel.queue_declare(queue=QUEUE_NAME, durable=True)

message = ' '.join(sys.argv[1:]) or "Hello World from Python!"

channel.basic_publish(
    exchange='',
    routing_key=QUEUE_NAME,
    body=message,
    properties=pika.BasicProperties(
        delivery_mode=pika.spec.PERSISTENT_DELIVERY_MODE,
    )
)

print('[PY] Sent %r' % message)
connection.close()

File: WorkQueues/js/sender/src/new_task.js
var amqp = require('amqplib/callback_api');

const queue_name = "task_queue";

amqp.connect("amqp://rabbitmq", function(error0, connection) {
    if(error0) {
        throw error0;
    }

    connection.createChannel(function(error1, channel) {
        if(error1) {
            throw error1;
        }

        const msg = process.argv.slice(2).join(' ') || "Hello World from Javascript!";

        channel.assertQueue(queue_name, {
            durable: true
        });

        channel.sendToQueue(queue_name, Buffer.from(msg), {
            persistent: true
        });
        
        console.log("[JS] Sent '%s'", msg);
    });

    setTimeout(function() {
        connection.close();
        process.exit(0);
    }, 500);
});